app/controllers/registration_tables_controller.rb

```diff
@@ -84,25 +86,42 @@ def create
     @registration_table = RegistrationTable.new(registration_table_params)
     respond_to do |format|
       if @registration_table.save
-        format.html { redirect_to [@event], notice: 'Table was successfully added.' }
-        format.json { render :show, status: :created, location: [@event] }
+        if @registration_table.payment_ok?
+          format.html {redirect_to [@event], notice: 'Table was successfully added.'}
+        else
+          format.html {redirect_to new_event_session_table_registration_table_table_payment_path(@event, @session, @table, @registration_table), notice: PREMIUM_MESSAGE}
```

Comment:
Oh, trust me, its "fun". This is railsy stuff, which I am probably not doing the best of. There is a way to collapse the length of that, but yeah, this is using the routes "properly"